### ISSUE TITLE
unable to launch on native wayland

### ISSUE THREAD

**Issue by UltraBlackLinux** [EvilSupahFly/Amulet-Flatpak#1]
Hey there,
glad everything has worked out :)
I'm running into some issues though:
```
INFO - PyMCTranslate Version 291
Unable to access the X Display, is $DISPLAY set properly?
```
if I force run it with xwayland (`env -u WAYLAND_DISPLAY flatpak run ...`) it works. 
Any idea why?
Thanks!

**Comment by EvilSupahFly** [comment]
That's strange because my initial production machine was running Wayland. I'll definitely take a look when I get home tonight though. What distribution are you running?

**Comment by EvilSupahFly** [comment]
Also, are you on KDE?

**Comment by UltraBlackLinux** [comment]
Nope, I'm on hyprland

Arch Linux, as mentioned in #2

**Comment by EvilSupahFly** [comment]
Sorry, I misread your name and thought you were different people. 😅

**Comment by EvilSupahFly** [comment]
When I re-built the Flatpak, now based on Amulet 0.10.35, I made sure to include support for X11 and Wayland both, so it launches without issue, but it still takes a nosedive like the one you experienced in #2, so that's my next task.

**Comment by UltraBlackLinux** [comment]
Sorry to disappoint you but I'm still running into issues with `Unable to access the X Display, is $DISPLAY set properly?`
Furthermore, running it within xwayland, the amulet process closes randomly during almost every step of the resource pack download and install process. I cannot get past the loading screen at all.

Here's the full log:
```
INFO - PyMCTranslate Version 291
INFO - Loading level /home/foo/.local/share/PrismLauncher/instances/1.20.2 for real/.minecraft/saves/§3⚀§b Random Royale §3⚅§8 v1.1.0
INFO - Loading level /home/foo/.local/share/PrismLauncher/instances/1.20.2 for real/.minecraft/saves/§3⚀§b Random Royale §3⚅§8 v1.1.0
Inconsistent value (1) for DRI_PRIME. Should be < 1 (GPU devices count). Using: 0
INFO - Downloading java launcher manifest file.
INFO - Downloading java launcher manifest file.
INFO - Finished downloading java launcher manifest file.
INFO - Finished downloading java launcher manifest file.
```

Edit: Just stumbled across this issue here: https://github.com/wxWidgets/Phoenix/issues/2557
Apparently wayland is unsupported.
If you remove the wayland and wayland-fallback options within the amulet manifest you can just force X11.

**Comment by EvilSupahFly** [comment]
I'm running Wayland on my production machine and it ran fine. After work tonight I'll boot up a couple of VMs for different distributions and see what comes up in the logss

**Comment by EvilSupahFly** [comment]
I've rebuilt the Flatpak with a modified `amulet.yml` and I didn't encounter any issues aside from some of the textures being broken (see [Amulet #1091](https://github.com/Amulet-Team/Amulet-Map-Editor/issues/1091)).

![Screenshot from 2024-08-25 19-06-45](https://github.com/user-attachments/assets/6d1e21e6-20d6-4499-885b-e15380be7b92)
![Screenshot from 2024-08-25 19-00-45](https://github.com/user-attachments/assets/77a1e2ec-4bc5-4460-925f-ce325a9d284a)

Before you try the new Flatpak from the releases section, be sure to uninstall your current Flatpak version:
```
flatpak uninstall com.github.amulet_map_editor
```

Also, as you can [see here](https://pastebin.com/CTA8mqrb), I ran the flatpak with `-vvv` to get a crap-ton of output. When you run it, try that too, and with all the extra detail, hopefully we'll see specifically where it's breaking down for you.

```
flatpak run -vvv com.github.amulet_map_editor
```

**Comment by UltraBlackLinux** [comment]
yep that seems to be working mostly!

As you've already noted, quite a few textures are missing, but that's progress :)

Annoyingly all buttons inside the 3D view are invisible and flicker in and out of view once when clicking inside the 3D view, which is quite unusable still...
~~this should probably go into a new issue shouldn't it~~

**Comment by EvilSupahFly** [comment]
The thing with the buttons has been an issue on Linux since the earliest days of Amulet.

The thing with the textures is new.

 ¯⁠\⁠_⁠(⁠ ͡⁠°⁠ ͜⁠ʖ⁠ ͡⁠°⁠)⁠_⁠/⁠¯

**Comment by tibequadorian** [comment]
I still have this issue. Running the flatpak in wayland doesn't work

**Comment by UltraBlackLinux** [comment]
logs please

**Comment by EvilSupahFly** [comment]
> logs please

Can you show your runtime logs? On my Endeavour test machine (Arch-based, uses Wayland), it runs fine - except for the missing textures.

**Comment by EvilSupahFly** [comment]
It's possible to run the Flatpak on Wayland if you have Xwayland. However, until the foundations in wxPython are patched, this will continue to be an issue. I'll update the README.

**Comment by EvilSupahFly** [comment]
Good news for Wayland users!

While I (still!!!) haven't figured out why Water and Lava textures won't load (though I think I'm close), I have fixed the Wayland issue, though it's really just a band-aid solution until wxPython itself is fixed.


By inserting `wx.PyApp.IsDisplayAvailable = lambda _: True` [here](https://github.com/Amulet-Team/Amulet-Map-Editor/blob/128de1caec6cc7da035b8336cd804d33aa3d5adc/amulet_map_editor/__main__.py#L48
), I've resolved the issue with the Flatpak version of Amulet not running on Wayland.

However, on my test rig, Amulet's fonts are really garbage looking. I don't know if it's an issue with my setup, or a side effect of monkey-patching Amulet's wxPython calls.

At least it's progress, right?
